pkg/k8s/watchers/pod.go

```diff
@@ -475,6 +485,14 @@ func (k *K8sWatcher) deleteK8sPodV1(pod *slim_corev1.Pod) error {
 		k.redirectPolicyManager.OnDeletePod(pod)
 	}
 
+	podNSName := k8sUtils.GetObjNamespaceName(&pod.ObjectMeta)
+	podEP := k.endpointManager.LookupPodName(podNSName)
+	if podEP != nil && podEP.NoTrackPort != "" {
+		podEP.UpdateNoTrackRules(func(ns, podName string) (noTrackPort string, err error) {
+			return "", nil
+		})
```

Comment:
Also worth noting here, `UpdateNoTrackRules()` blocks on the channel for completion of the logic. Not sure exactly how the k8s informer in the parent function of the current function would behave if this blocks for an indefinite period (say if the endpoint eventqueue has a lot of processing to do and we're starved for CPU by a limiter). I don't think this is a super common case, but the reason I bring it up is that I'm not seeing other logic in this function that blocks for a potentially long period, so it could end up having unintended repercussions.

Reply:
^ Maybe not so much of a concern given that the corresponding eventqueue interactions in the update function would have the same behaviour, so I don't think there's any explicit changes you need to make here based on this particular point.

Reply:
Removed this logic here.